Utilities.Math.Functions.fallingFactorial fails to translate in JModelica

The function ```IBPSA/Utilities/Math/Functions/fallingFactorial.mo``` fails to translate in the latest JModelica with the error
```
*** Compiling IBPSA.Utilities.Math.Examples.FallingFactorial
sources/IBPSA_Utilities_Math_Examples_FallingFactorial_funcs.c: In function 'func_IBPSA_Utilities_Math_Functions_fallingFactorial_def0':
sources/IBPSA_Utilities_Math_Examples_FallingFactorial_funcs.c:13:16: error: 'INT_MAX' undeclared (first use in this function)
     maxInt_v = INT_MAX;
                ^
sources/IBPSA_Utilities_Math_Examples_FallingFactorial_funcs.c:13:16: note: each undeclared identifier is reported only once for each function it appears in
make[1]: *** [IBPSA_Utilities_Math_Examples_FallingFactorial_funcs.o] Error 1
```
The following change fixes the model:
```
--- a/IBPSA/Utilities/Math/Functions/fallingFactorial.mo
+++ b/IBPSA/Utilities/Math/Functions/fallingFactorial.mo
@@ -5,10 +5,8 @@ function fallingFactorial "Returns the k-th falling factorial of n"
   input Integer n "Integer number";
   input Integer k "Falling factorial power";
   output Integer f "k-th falling factorial of n";
-
 protected
-  Integer maxInt = 2147483647 "Max 32-bit integer";
-
+  Integer maxInt = 2147483646 "Max 32-bit integer";
 algorithm
```
Note that
```
--- a/IBPSA/Utilities/Math/Functions/fallingFactorial.mo
+++ b/IBPSA/Utilities/Math/Functions/fallingFactorial.mo
@@ -5,10 +5,9 @@ function fallingFactorial "Returns the k-th falling factorial of n"
   input Integer n "Integer number";
   input Integer k "Falling factorial power";
   output Integer f "k-th falling factorial of n";
-
 protected
-  Integer maxInt = 2147483647 "Max 32-bit integer";
-
+  Integer maxInt = 2147483646 "Max 32-bit integer";
+  Integer test = Modelica.Constants.Integer_inf "test";
 algorithm

```
fails with
```
*** Compiling IBPSA.Utilities.Math.Functions.Examples.FallingFactorial
sources/IBPSA_Utilities_Math_Functions_Examples_FallingFactorial_funcs.c: In function 'func_IBPSA_Utilities_Math_Functions_fallingFactorial_def0':
sources/IBPSA_Utilities_Math_Functions_Examples_FallingFactorial_funcs.c:15:14: error: 'INT_MAX' undeclared (first use in this function)
     test_v = INT_MAX;
              ^
```
and hence I will hard code the upper limit.

--- a/IBPSA/Utilities/Math/Functions/fallingFactorial.mo
+++ b/IBPSA/Utilities/Math/Functions/fallingFactorial.mo
@@ -1,14 +1,11 @@
 within IBPSA.Utilities.Math.Functions;
 function fallingFactorial "Returns the k-th falling factorial of n"
   extends Modelica.Icons.Function;
-
   input Integer n "Integer number";
   input Integer k "Falling factorial power";
   output Integer f "k-th falling factorial of n";
-
 protected
-  Integer maxInt = 2147483647 "Max 32-bit integer";
-
+  Integer maxInt = 2147483646 "Max 32-bit integer";
 algorithm
   if k > n then
     f := 0;
@@ -26,6 +23,12 @@ Function that evaluates the falling factorial \"k-permutations of n\".
 </p>
 </html>", revisions="<html>
 <ul>
+<li>
+May 22, 2019, by Michael Wetter:<br/>
+Changed upper limit for <code>maxInt</code>.<br/>
+This is for
+<a href=\"https://github.com/ibpsa/modelica-ibpsa/issues/1145\">IBPSA, issue 1145</a>.
+</li>
 <li>
 February 9, 2018, by Massimo Cimmino:<br/>
 First implementation.

